### PR TITLE
Milestone : Catalog of my things

### PR DESCRIPTION
**Pull request by bronnie313**
## Catalog of my things app
- Added linter configuration files
- Added all methods visible in the diagram
- In this milestone of the project, I did the following:
- Created Item class
- Added methods for Item class
- Created a Book class
- Created a Label class
- Define and set up properties for Book class
- Define and set up properties for Label class
- Implemented all methods
- Added unit tests for implemented methods
- Created a MusicAlbum class
- Created a Genre class with an association with the Item class
- Implemented methods in the Genre class
- Implemented methods in this MusicAlbum class
- Added unit test for all implemented methods
- The App lists all music albums
- It lists all genres
- Adds music album
- Data is preserved in the .json files
- Created schema.sql file with tables
- Added options (List all books, List all labels, and Add a book)
- Preserved data by saving collections in .json files
- Created a schema.sql file with the appropriate tables
- Create Game and Author class
- Define all Game and Author class properties
- Add unit test for Game and Author class
- Preserve all data
- These options are available (List of games, List all authors, Add a game)
- Create database tables for Game and Author

https://github.com/bronnie313/ruby-capstone-project/assets/37089236/1381c801-7637-4c39-bad2-78a08957f63a

### CODE REVIEW

File: game.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  def add_author(author)
+    return if authors.include?(author)
+
+    authors << author
+    author.add_item(self)
+  end
+
+  def can_be_archived?
+    super || (Date.today - Date.parse(@last_played_at) > 365 * 2)
+  end
```
Comment (FrederickMih) [inline game.rb:47]: - [ ] I don't think this method is doing exactly what was expected. This method should return true if the parent method(super) returns true **AND** if `last_played_at ` is **older** than 2 years.

Let's kindly check my piece of code below and share your thoughts :smile: 

```
current_year = Time.now.year
2_years_ago = current_year - 2
super && @last_played_at < 2_years_ago
```

File: README.md [hunk 2]
```diff
@@ -1 +1,172 @@
...
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+<div align="center"><hr width="250px"/></div>
+
+# 📨 Catalog of my things <a name="about-project"></a>
```
Comment (FrederickMih) [inline README.md:31]: - [ ] Based on the requirements, the readme should contain instructions on how to setup and start the app locally. Let's kindly include these instructions in the readme file.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @bronnie313, @WilhelmK109, @peter-akhigbe , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- [ ] Data was preserved in the JSON files on exit :+1: 
- [ ] PR is professional :+1: 



## Required Changes ♻️

- [ ] Nice job presenting the features of your app. Once more let's kindly adhere to the requirements of the project. The maximum length of the project's presentation is supposed to be 5 minutes. Let's kindly make sure the video contains a highlight about a piece of code you found interesting or proud of implementing. I'll kindly suggest that each member of the project share his IDE while presenting his part of the project. 

_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by peter-akhigbe** [comment]
Updated Presentation Video:

https://github.com/bronnie313/ruby-capstone-project/assets/12002619/9d7dc8cc-d717-4a33-a467-e03e0a1f3382

**Review by rotimiazeez — APPROVED** [review]
Hi TEAM 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Good video presentation 👍🏽
* Good teamwork ✌🏽
* App working as expected 💯
* All required functionalities are implemented 👏🏽
* Descriptive PR and README file 👌🏽

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**